requirements.txt

```diff
@@ -1,6 +1,35 @@
-requests
-pluginbase
-jinja2
-enum34; python_version < '3.4'
-javalang
-click
\ No newline at end of file
+certifi==2018.1.18 \
+    --hash=sha256:14131608ad2fd56836d33a71ee60fa1c82bc9d2c8d98b7bdbc631fe1b3cd1296 \
+    --hash=sha256:edbc3f203427eef571f79a7692bb160a2b0f7ccaa31953e99bd17e307cf63f7d
+chardet==3.0.4 \
+    --hash=sha256:84ab92ed1c4d4f16916e05906b6b75a6c0fb5db821cc65e70cbd64a3e2a5eaae \
+    --hash=sha256:fc323ffcaeaed0e0a02bf4d117757b98aed530d9ed4531e3e15460124c106691
+click==6.7 \
+    --hash=sha256:29f99fc6125fbc931b758dc053b3114e55c77a6e4c6c3a2674a2dc986016381d \
+    --hash=sha256:f15516df478d5a56180fbf80e68f206010e6d160fc39fa508b65e035fd75130b
+enum34==1.1.6 ; python_version < "3.4" \
```

Comment:
I see what you did there. Nice hack.

For the records, that simplifies the following line so we don't need a special case for the  `python_version < "3.4"`:
```
required = [requirement.split(' ')[0] for requirement in all_lines if not requirement.startswith(' ')]
```

We should still add an entry into our not yet created FAQ or README to explain that people should upgrade their pip and setuptools if they hit the `Invalid environment marker: python_version<"3.4"` issue.